### PR TITLE
*: move the `Handle` of stats to single package.

### PR DESCRIPTION
**Pull request by winoros**
### What problem does this PR solve? 

Solve the cycle import problem in #10106 

### What is changed and how it works?

Split the `statistics` package to two parts
```
statistics
|
└-handler
```

Then `statistics` won't import `infoschema` package. So the cycle import is resolved.

### Check List 

Tests 

 - Unit test

Code changes

 - Has exported function/method change
 - Has exported variable/fields change

### CODE REVIEW

File: statistics/builder.go [hunk 1]
```diff
@@ -65,11 +65,11 @@ func (b *SortedBuilder) Iterate(data types.Datum) error {
 		b.hist.Buckets[b.bucketIdx].Count++
 		b.hist.Buckets[b.bucketIdx].Repeat++
 	} else if b.hist.Buckets[b.bucketIdx].Count+1-b.lastNumber <= b.valuesPerBucket {
-		// The bucket still have room to store a new item, update the bucket.
+		// The bucket still have room to store a new item, Update the bucket.
```
Comment (lamxTyler) [inline statistics/builder.go]: Seems like we should not update the comment in this file.

File: statistics/feedback.go [hunk 2]
```diff
@@ -494,20 +501,20 @@ const (
 	maxBucketFraction = 1 / 10.0
 )
 
-// getBucketScore gets the score for merge this bucket with previous one.
+// getBucketScore gets the score for Merge this bucket with previous one.
```
Comment (lamxTyler) [inline statistics/feedback.go]: And also these comments in the file.

File: statistics/handler/dump.go [hunk 3]
```diff
@@ -0,0 +1,364 @@
...
+				Count:      count,
+				IsHandle:   tableInfo.PKIsHandle && mysql.HasPriKeyFlag(colInfo.Flag),
+			}
+			tbl.Columns[col.ID] = col
+		}
+	}
+	return tbl, nil
+}
+
+func (h *Handle) dumpRangeFeedback(sc *stmtctx.StatementContext, ran *ranger.Range, rangeCount float64, q *statistics.QueryFeedback) error {
```
Comment (lamxTyler) [inline statistics/handler/dump.go]: This file should be for `dump stats`. Maybe it is better to move it to another file?

File: statistics/selectivity.go [hunk 4]
```diff
@@ -192,9 +192,9 @@ func (coll *HistColl) Selectivity(ctx sessionctx.Context, exprs []expression.Exp
 			if err != nil {
 				return 0, nil, errors.Trace(err)
 			}
-			nodes = append(nodes, &StatsNode{Tp: colType, ID: id, mask: maskCovered, Ranges: ranges, numCols: 1})
-			if colInfo.isHandle {
-				nodes[len(nodes)-1].Tp = pkType
+			nodes = append(nodes, &StatsNode{Tp: ColType, ID: id, mask: maskCovered, Ranges: ranges, numCols: 1})
+			if colInfo.IsHandle {
```
Comment (xiaoronglv) [inline statistics/selectivity.go]: Hi @winoros 

Could you please explain the meaning of `IsHandle`?  

Does that mean colInfo with `true` value is the primary key?

Thanks

**Pull request merged**